### PR TITLE
Api error logging

### PR DESCRIPTION
**Pull request by yhua045**
Two questions:
1. I manually created a 'log' directory under the root folder. Will this folder be deployed automatically to the server when I run deploy:qa or do I need to include additional script to handle the directory creation?
2. Is including the 'error log' at the same directory as the web server the standard practice? or should it be placed at some where else?

I also moved the error logging to the middleware to reduce the duplicated code.

### CODE REVIEW

**Comment by yhua045** [comment]
[(3) API - error logging [3]](https://trello.com/c/hT59kzzf/128-3-api-error-logging-3)

**Comment by Josephu** [comment]
normally the trick to keep an empty folder is to add a file in the folder eg. `.gitkeep` so the folder can be put into git. If you are using winston, does it automatically create folder? If it does, then we don't need to worry about it.

I think it is fine to put logs under the same directory as the application. I like it that way actually.

Yeah, thank you for extracting it as a middleware. :)

File: package.json [hunk 1]
```diff
@@ -33,7 +33,8 @@
     "sinon": "^4.1.2",
     "sql.js": "0.3.2",
     "supertest": "^3.0.0",
-    "winston": "^2.4.0"
+    "winston": "^2.4.0",
+    "winston-daily-rotate-file": "^1.7.2"
```
Comment (Josephu) [inline package.json]: nice :)

File: log/.gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+*
+!.gitignore
```
Comment (Josephu) [inline log/.gitignore]: You placed .gitignore already, that will keep the folder :)

**Pull request merged**